AmplifyPlugins/Predictions/AWSPredictionsPlugin/Support/Internal/NativeWebSocketProvider.swift

```diff
@@ -0,0 +1,143 @@
+//
+// Copyright 2018-2019 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import AWSTranscribeStreaming
+
+class NativeWebSocketProvider: NSObject, AWSTranscribeStreamingWebSocketProvider, URLSessionWebSocketDelegate {
+
+    //swiftlint:disable weak_delegate
+    var clientDelegate: AWSTranscribeStreamingClientDelegate
+    var webSocketTask: URLSessionWebSocketTask!
+    var urlSession: URLSession!
+    let delegateQueue = OperationQueue()
+    var callbackQueue: DispatchQueue!
+    private var pingTimer: Timer?
+
+    init(clientDelegate: NativeWSTranscribeStreamingClientDelegate, callbackQueue: DispatchQueue) {
+        self.clientDelegate = clientDelegate
+        self.callbackQueue = callbackQueue
+        super.init()
+        self.urlSession = URLSession(configuration: .default, delegate: self, delegateQueue: delegateQueue)
+
+    }
+
+    func configure(with urlRequest: URLRequest) {
+        if let url = urlRequest.url {
+            webSocketTask = urlSession.webSocketTask(with: url)
+        }
+    }
+    //required by protocol but not needed
+    func setDelegate(_ delegate: AWSTranscribeStreamingClientDelegate, dispatchQueue: DispatchQueue) {
+
+    }
+
+    func urlSession(_ session: URLSession,
+                    webSocketTask: URLSessionWebSocketTask,
+                    didOpenWithProtocol protocol: String?) {
+        let status = AWSTranscribeStreamingClientConnectionStatus.connected
+
+        callbackQueue.async {
+            self.clientDelegate.connectionStatusDidChange(status, withError: nil)
+        }
+    }
+
+    func urlSession(_ session: URLSession,
+                    webSocketTask: URLSessionWebSocketTask,
+                    didCloseWith closeCode: URLSessionWebSocketTask.CloseCode,
+                    reason: Data?) {
+        let status = AWSTranscribeStreamingClientConnectionStatus.closed
+        let error = NSError(domain: AWSTranscribeStreamingClientErrorDomain, code: closeCode.rawValue, userInfo: nil)
+
+        callbackQueue.async {
+            self.clientDelegate.connectionStatusDidChange(status, withError: error)
+        }
+    }
+
+    func connect() {
+        // required to open socket
+        webSocketTask.resume()
+        listen()
+    }
+
+    func disconnect() {
+        webSocketTask.cancel(with: .normalClosure, reason: nil)
+        pingTimer?.invalidate()
```

Comment:
Can we remove the ping logic if its not required?

Reply:
removed